examples/jsm/nodes/lighting/AnalyticLightNode.js

```diff
@@ -191,6 +191,7 @@ class AnalyticLightNode extends LightingNode {
 		rtt.setSize( light.shadow.mapSize.width, light.shadow.mapSize.height );
 
 		light.shadow.updateMatrices( light );
+		light.shadow.camera.layers.mask = camera.layers.mask;
```

Comment:
I like a lot this solution, currently the `WebGLRenderer` only relies on the main camera and never care about the `shadowCamera.layers`:
https://github.com/mrdoob/three.js/blob/b38ef166c7ed9efa1cac71b4fd16d0e168079ad7/src/renderers/webgl/WebGLShadowMap.js#L335
but I believe this new solution is as good if not better!

Reply:
An interesting point about WebGPURenderer is that we do not have a shadow class directly linked to the core, but it is part of the Node class integrated with the lights, allowing different approaches to shadow rendering as extensions.